### ISSUE TITLE
[Regression] RPCS3 stuck on Compiling Modules with SPU Mega (#15429)

### ISSUE THREAD

**Issue by gamrXerus** [RPCS3/rpcs3#15894]
### Quick summary

Ever since PR#15429 Call of Duty Black Ops 2 gets stuck on Compiling PPU Modules. If I restart the emulator to bypass it, the game wont get past the dolby logo. 

### Build with regression

PR #15429

### Attach two log files

Black Ops 2 working: version 0.0.32-16500 [RPCS3.log](https://github.com/user-attachments/files/16488362/RPCS3.log)
Black Ops 2 working: version 0.0.32-16500 [RPCS3.log.gz](https://github.com/user-attachments/files/16488364/RPCS3.log.gz)
Black Ops 2 not working: version 0.0.32-16505 [RPCS3.log](https://github.com/user-attachments/files/16488374/RPCS3.log)
Black Ops 2 not working: version 0.0.32-16505  [RPCS3.log.gz](https://github.com/user-attachments/files/16488375/RPCS3.log.gz)
Black Ops 2 master: version 16746:  [RPCS3.log](https://github.com/user-attachments/files/16488378/RPCS3.log)
Black Ops 2 master: version 16746: [RPCS3.log.gz](https://github.com/user-attachments/files/16488380/RPCS3.log.gz)

### System configuration

Operating System: Fedora 40 (KDE Plasma)
CPU: Intel i9 11900K
GPU: AMD Radeon RX 6800
Driver: Mesa 24.1.5

**Comment by alexespinoza28** [comment]
If you look on the rpcs3 compatibility page it shows that Black Ops 2 is currently in the "in-game phase". In-game phase means Games that either can't be finished, have serious glitches, or have insufficient performance. Therefore, this game has too many issues to pinpoint your problem and it will not be playable anytime soon.

**Comment by gamrXerus** [comment]
Changing the setting SPU Block Size from "Mega" to "Safe" allowed me to get in game with the latest version of rpcs3. Apparently newer builds need more safe settings in order for Call of Duty to work.

**Comment by elad335** [comment]
Still an issue?

**Comment by gamrXerus** [comment]
> Still an issue?

Yes it is. I just tested the latest rpcs3 appimage (v0.0.32-16874) and deleted my cache folder manually this time( /home/username/.cache/rpcs3/). Then I ran the game using Mega and Safe. It still does not work with Mega unfortunately.

**Comment by AniLeo** [comment]
Needs retesting on latest with a new log file

**Comment by gamrXerus** [comment]
@AniLeo I just tested with the latest rpcs3 (v0.0.36-17760) and the problem seems to be fixed. Here is my log file.

[RPCS3.log](https://github.com/user-attachments/files/19636840/RPCS3.log)   Great work by the RPCS3 team.

**Comment by gamrXerus** [comment]
@AniLeo I just tested the latest rpcs3 appimage (v0.0.36-17760) and it is still an issue with SPU Mega. See attached log file 

[RPCS3.log](https://github.com/user-attachments/files/19636946/RPCS3.log)

**Comment by digant73** [comment]
@gamrXerus is that still an issue?

**Comment by LucioXerus** [comment]
> [@gamrXerus](https://github.com/gamrXerus) is that still an issue?

Yes. I just tested with the latest release and this is still an issue

**Comment by digant73** [comment]
not sure if mega, giga are considered by main developers as always to be safe. AFAIR, giga is causing a crash on several games such as KZ3, KZ3, R3 etc.

**Comment by AniLeo** [comment]
Needs testing with #17742

**Comment by elad335** [comment]
Try disabling `PPU/SPU LLVM Precompilation` in advanced tab and clear SPU Cache.

**Comment by LucioXerus** [comment]
@AniLeo @elad335  I disabled PPU/SPU LLVM Precompilation, cleared my cache and ran the game with pr17742. I was able to get the game to start. The second intro video wont play with Mega enabled tho. Should I close this issue?

**Comment by AniLeo** [comment]
If the video plays with SPU Block Size Safe but not SPU Block Size Mega, I would say to keep this open

**Comment by LucioXerus** [comment]
Ok I'll leave it open then

**Comment by elad335** [comment]
I tested it, it does complete if you wait about 20 seconds on the last SPU function. Though it is an inducation that I need to optimize constant propagation.

**Comment by AniLeo** [comment]
> need to optimize constant propagation.

Thanks, don't forget to open a ticket so it's kept in mind

Closing since the original bug for this ticket was solved

**Comment by LucioXerus** [comment]
I can confirm pr #17771 fixes this issue.
